stages/07.export-stage/03.export-image/configure-setup.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# kuiper2.0 - Embedded Linux for Analog Devices Products
+#
+# Copyright (c) 2024 Analog Devices, Inc.
+# Author: Larisa Radu <larisa.radu@analog.com>
+
+BOOTLOADER_DEV=$1
+
+if [ "${CONFIG_XILINX_INTEL_BOOT_FILES}" = y ]; then
+	if [[ ! -z "${PROJECT_NAME}"  && ! -z "${BOARD}" ]]; then
+	
+		# Extract project description from kuiper.json by selecting the project set in the configuration file
+		PROJECT_DESCRIPTION=$(cat ${BUILD_DIR}/boot/kuiper.json \
+							| jq '."projects"' \
+							| jq '.[] | select(.name=="'${PROJECT_NAME}'" and .board=="'${BOARD}'")')
+		
+		# Check if project exists
+		if [[ -z "${PROJECT_DESCRIPTION}" ]]; then
+			echo "Cannot find project ${PROJECT_NAME} for board ${BOARD}. Setup not configured."
+		else
+			# Extract kernel path from the project description and copy it to Kuiper boot directory
+			cp ${BUILD_DIR}$(sed -n "/kernel/p" <<< "${PROJECT_DESCRIPTION}" | cut -d'"' -f4) ${BUILD_DIR}/boot
+			
+			# Extract paths with boot files from the project description
+			paths=$(sed -n "/path/p" <<< "${PROJECT_DESCRIPTION}" | cut -d'"' -f4)
+			
+			# Add the correct prefix for all paths and copy them to Kuiper boot directory
+			paths=${paths//\/boot/${BUILD_DIR}\/boot}
+			cp $paths ${BUILD_DIR}/boot
+			
+			# Check if project platform is Intel
+			if [[ ! -z $(sed -n "/intel/p" <<< "${PROJECT_DESCRIPTION}") ]]; then
+			
+				# Move copied extlinux.conf to extlinux folder
+				mv ${BUILD_DIR}/boot/extlinux.conf ${BUILD_DIR}/boot/extlinux/
+				
+				# Write preloader file to corresponding image partition
+				dd if=${BUILD_DIR}$(sed -n "/preloader/p" <<< "${PROJECT_DESCRIPTION}" | cut -d'"' -f4) of=${BOOTLOADER_DEV}
```

Comment:
Please prepare the preloader file in a separate line and write it on the next line.
Probably you'll need to 'dd' with sudo, and you can add some other 'dd' options, like block size, show progress, verbosity etc. 
Check also dd exit code / output errors.